src/sys/stat.rs

```diff
@@ -402,12 +402,87 @@ pub fn lutimes<P: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
+/// Timestamp update operation specifier used in [`utimensat`] and [`futimens`].
+#[cfg(not(target_os = "redox"))] // Redox does not support `UTIME_OMIT` and `UTIME_NOW`
+#[derive(Copy, Clone, Debug)]
+pub enum TimestampSpec {
+    /// Leave the timestamp unchanged.
+    Omit,
+    /// Update the timestamp to `Now`
+    Now,
+    /// Update the timestamp to the specific value specified in `TimeSpec`.
+    Set(TimeSpec),
+}
+
+/// A helper function to construct an instance of `libc::timespec` according to
+/// the operation specified in `spec`.
+#[cfg(not(target_os = "redox"))] // Redox does not support `UTIME_OMIT` and `UTIME_NOW`
+fn create_timespec(spec: &TimestampSpec) -> libc::timespec {
```

Comment:
I know it's only an internal function, so this is no big deal.  But it would be a little more ergonomic to use if this were written as
```rust
impl From<TimestampSpec> for libc::timespec {...}
```